src/agent/src/netlink.rs

```diff
@@ -312,7 +312,11 @@ impl Handle {
 
         for route in list {
             let link = self.find_link(LinkFilter::Name(&route.device)).await?;
-            let is_v6 = is_ipv6(route.get_gateway()) || is_ipv6(route.get_dest());
+            let is_v6 = if route.get_gateway().is_empty() && route.get_dest().is_empty() {
+                route.get_family() == IPFamily::v6
```

Comment:
Is there any reason we wouldn't want to use get_family directly, regardless if the gateway and destination are not provided?  Are we populating the family as part of the gRPC regardless?

Reply:
This is for backward compatibility purposes, unless we're ok with saying the new agent will only work with the corresponding  runtime changes.

Reply:
@egernst I changed my mind and went the cleaner way: We only rely on the Route family field now for switching between v4 and v6. Running this new agent means you'll have to run the fixed runtime as well, but I believe this is acceptable.